test/inductor/test_compiled_autograd.py

```diff
@@ -376,6 +376,170 @@ def eager_check():
 
             eager_check()
 
+    def test_custom_fn_saved_tensors(self):
+        def fn():
+            class MySin(torch.autograd.Function):
+                @staticmethod
+                def forward(ctx, x):
+                    ctx.save_for_backward(x)
+                    return torch.sin(x)
+
+                @staticmethod
+                def backward(ctx, gO):
+                    (x,) = ctx.saved_tensors
+                    return gO * torch.cos(x)
+
+            for i in [10, 100, 10]:
+                x = torch.arange(0.0, i, requires_grad=True)
+                out = MySin.apply(x)
+                loss = out.sum()
+                loss.backward()
+                yield x.grad
+
+        self.check_output_and_recompiles(fn, 2)
+
+    def test_custom_fn_saved_multiple_tensors(self):
+        def fn():
+            class MyFn(torch.autograd.Function):
+                @staticmethod
+                def forward(ctx, x):
+                    ctx.save_for_backward(x, x)
+                    return torch.sin(x)
+
+                @staticmethod
+                def backward(ctx, gO):
+                    (x1, x2) = ctx.saved_tensors
+                    return gO * torch.cos(x1) * torch.cos(x2)
+
+            for i in [10, 100, 10]:
```

Comment:
```suggestion
            for i in [10, 100, 10, 15, 20, 25]:
```

Should stil be 2 recompiles because we auto-mark things dynamic.